internal/pkg/database/migrate.go

```diff
@@ -0,0 +1,128 @@
+package database
+
+import (
+	"database/sql"
+	"errors"
+	"log"
+
+	"fmt"
+	"github.com/jmoiron/sqlx"
+)
+
+// TODO: pick correct type for timestamp (with/without timezone)
+var migrationSchema = `
+  CREATE TABLE IF NOT EXISTS migrations (
+    name text UNIQUE,
+    timestamp timestamp
+  )
+`
+
+type migrationFunc func(Queryer) error
+
+type migration struct {
+	name string
+	fn   migrationFunc
+}
+
+type migrationTracker struct {
+	Name      string `db:"name"`
+	Timestamp string `db:"timestamp"`
+}
+
+func migrate(db *sqlx.DB, migrations []migration) error {
+	err := createMigrationTable(db)
+	if err != nil {
+		return fmt.Errorf("unable to create migration table: %w", err)
+	}
+
+	for _, m := range migrations {
+		err = applyMigration(db, m.name, m.fn)
+		if err != nil {
+			return fmt.Errorf("unable to apply migration '%s': %w", m.name, err)
+		}
+	}
+
+	return nil
+}
+
+func createMigrationTable(db *sqlx.DB) error {
```

Comment:
This doesn't seem worth the dedicated func, as it's mostly error checking which you do after you call it as well.

Reply:
I tried removing it, and we could, but I rather not, since this function is quite handy when testing the code. The function is used to create the migrationtable when we're testing other migration functions.